src/_posts/2020-04-28-terraform-with-multiple-environments.md

```diff
@@ -0,0 +1,167 @@
+
+
+    - environments
+    |- dev
+    |  |- main.tf
+    |  |- variables.tf
+    |  |- outputs.tf
+    |  |- dev.env.tfvars
+    |- test
+    |  |- main.tf -> ../dev/main.tf
+    |  |- variables.tf -> ../dev/variables.tf
+    |  |- outputs.tf -> ../dev/outputs.tf
+    |  |- test.env.tfvars
+    |- prod
+    |  |- main.tf -> ../dev/main.tf
+    |  |- variables.tf -> ../dev/variables.tf
+    |  |- outputs.tf -> ../dev/outputs.tf
+    |  |- prod.env.tfvars
+    |  |- prod.unique.tf
+    
+Unlike with workspaces, it is very difficult to make mistakes here. After all you need to navigate to the correct folder before you can execute your terraform commands. And symlinks keep in check the number of files that you need to take care off (which otherwise would be a big downside of this approach).
+
+# The third way - Multiple accounts, multiple workspaces
+
+But, wait - you say - does that mean I can't use workspaces with multiple accounts? Well, of course not. The above approach was based on the fact that we were keeping the terraform state of each account within each account. But if you switch to a single centralised place, then you wouldn't have the issue, and therefore you could go with using workspaces on multiple accounts. And Terraform uses a big number of backends especifically for this: [Artifactory](https://www.terraform.io/docs/backends/types/artifactory.html), [Consul](https://www.terraform.io/docs/backends/types/consul.html), [etcd v2](https://www.terraform.io/docs/backends/types/etcd.html) and [etcd v3](https://www.terraform.io/docs/backends/types/etcdv3.html), some "random" [http rest](https://www.terraform.io/docs/backends/types/http.html), [swift](https://www.terraform.io/docs/backends/types/swift.html), [Postgres](https://www.terraform.io/docs/backends/types/pg.html) and their own [Terraform Enterprise](https://www.terraform.io/docs/backends/types/terraform-enterprise.html). 
+
+Can I use a single s3 backend with a different profile (a different account) than the rest of the system. Well, yes, the backend can use different credentials than the rest of the setup. In fact, I discovered recently that you can have multiple providers (mixing or multiple accounts of the same one), give them different aliases, and choose for each resource which provider to use.
+
+In fact, some very quick code here how to deal with it, using as a centralized state storage an [Azure](https://azure.microsoft.com/) Blob Storage (mostly because I did not access to the "proper" ones, but I had access to the Codurance Azure playground account).
```

Comment:
```suggestion
In fact, some very quick code here on how to deal with it, using a centralized state storage such as [Azure](https://azure.microsoft.com/) Blob Storage (mostly because I did not have access to the "proper" ones, but I had access to the Codurance Azure playground account).
```